Lib/test/test_os.py

```diff
@@ -1252,6 +1253,43 @@ def tearDownClass(cls):
         os.rmdir(support.TESTFN)
 
 
+class CurrentDirTests(unittest.TestCase):
+
+    def setUp(self):
+        base = os.path.abspath(support.TESTFN)
+        self.tmp_dir = base + '_dir'
+        self.tmp_lnk = base + '_lnk'
+
+    def test_getcwd(self):
+        # os.getcwd() always returns the dereferenced path
+        with support.temp_cwd(self.tmp_dir):
+            os.chdir(self.tmp_dir)
+            self.assertEqual(self.tmp_dir, os.getcwd())
+            os.symlink(self.tmp_dir, self.tmp_lnk, True)
+            os.chdir(self.tmp_lnk)
+            self.assertEqual(self.tmp_dir, os.getcwd())
+            with mock.patch.dict('os.environ', {'PWD': self.tmp_dir}):
+                self.assertEqual(self.tmp_dir, os.getcwd())
+            with mock.patch.dict('os.environ', {'PWD': self.tmp_lnk}):
+                self.assertEqual(self.tmp_dir, os.getcwd())
+            os.unlink(self.tmp_lnk)
```

Comment:
The link will be leaked in the case of failure. It is better to use `addCleanup()`. Add the following line just before creating a link.
```python
self.addCleanup(support.unlink, self.tmp_lnk)
```